src/lib/comms/sol-coap.c

```diff
@@ -662,8 +662,11 @@ pending_timeout_cb(void *data)
     struct pending_reply *reply = data;
     struct sol_coap_server *server = reply->server;
 
-    if (reply->cb((void *)reply->data, server, NULL, NULL))
-        return true;
+    if (reply->cb((void *)reply->data, server, NULL, NULL)) {
+        if (reply->freshness != PENDING_REPLY_FREE_ME) {
+            return true;
+        }
+    }
 
     sol_ptr_vector_remove(&server->pending, reply);
```

Comment:
... and it is expected that the reply will be freed by whoever changed ```reply->freshness``` from ```PENDING_REPLY_QUIET``` to ```PENDING_REPLY_PENDING```.